src/rgw/rgw_rados.h

```diff
@@ -2427,6 +2427,16 @@ class RGWRados
   const string& get_current_period_id() {
     return current_period.get_id();
   }
+
+  int get_zonegroups_by_api(map<string, RGWZoneGroup>& zonegroups_by_api) {
+    int ret = -1;
+    if (!current_period.get_id().empty()) {
+      zonegroups_by_api = current_period.get_map().zonegroups_by_api;
+      ret = 0;
+    }
+    return ret;
+  }
+
```

Comment:
reasonable, but here we are only checking the existence of  location_constraint ， then checking whether location_constraint match the region(is_zonegroup_api_exist). Maybe `is_zonegroup_api_exist` is more accurate.

repushed.